pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -346,31 +350,70 @@ def build_model(
             )
 
             if self.time_varying_intercept:
-                time_index = pm.Data(
-                    "time_index",
+                intercept_time_index = pm.Data(
+                    "intercept_time_index",
                     self._time_index,
                     dims="date",
                 )
-                intercept_dist = get_distribution(
+
+                intercept_distribution = get_distribution(
                     name=self.model_config["intercept"]["dist"]
                 )
-                intercept = create_time_varying_intercept(
-                    time_index,
-                    self._time_index_mid,
-                    self._time_resolution,
-                    intercept_dist,
-                    self.model_config,
+                base_intercept = intercept_distribution(
+                    name="base_intercept", **self.model_config["intercept"]["kwargs"]
+                )
+
+                intercept_latent_process = create_time_varying_gp_multiplier(
+                    name="intercept",
+                    dims="date",
+                    time_index=intercept_time_index,
+                    time_index_mid=self._time_index_mid,
+                    time_resolution=self._time_resolution,
+                    model_config=self.model_config,
+                )
+                intercept = pm.Deterministic(
+                    name="intercept",
+                    var=base_intercept * intercept_latent_process,
+                    dims="date",
                 )
             else:
                 intercept = create_distribution_from_config(
                     name="intercept", config=self.model_config
                 )
 
-            channel_contributions = pm.Deterministic(
-                name="channel_contributions",
-                var=self.forward_pass(x=channel_data_),
-                dims=("date", "channel"),
-            )
+            if self.time_varying_media:
+                media_time_index = pm.Data(
+                    "media_time_index",
+                    self._time_index,
+                    dims="date",
+                )
+
+                base_channel_contributions = pm.Deterministic(
+                    name="base_channel_contributions",
+                    var=self.forward_pass(x=channel_data_),
+                    dims=("date", "channel"),
+                )
+
+                media_latent_process = create_time_varying_gp_multiplier(
+                    name="media",
+                    dims="date",
+                    time_index=media_time_index,
+                    time_index_mid=self._time_index_mid,
+                    time_resolution=self._time_resolution,
+                    model_config=self.model_config,
+                )
+                channel_contributions = pm.Deterministic(
+                    name="channel_contributions",
+                    var=base_channel_contributions * media_latent_process[:, None],
```

Comment:
yeah this is valid. Right now `time_varying_prior` is structured such that it is centered on 1. Logic for that is:

```Python
phi, sqrt_psd = gp.prior_linearized(Xs=X[:, None] - X_mid)
hsgp_coefs = pm.Normal(f"{name}_hsgp_coefs", dims=hsgp_dims)
f = phi @ (hsgp_coefs * sqrt_psd).T
centered_f = f - f.mean(axis=0) + 1  # <-- this line
```

Due to the scale of `f`, it's unlikely that it goes below 0, but it can happen (by a very small amount).

Could we floor it to zero?

Reply:
I thought there was a transformation on the f to make positive?

Reply:
Guys, I was playing with this and I think the solution is to allow the distribution of the HSGP coefficients to be arbitrary, with default a only positive distribution. Basically follow the same pattern that we currently allow to all other distributions from the model. We could easily add to the "**_tvp_config", the keys "dist" and"kwargs".

The current setup give this prior for the HSGP:
![image](https://github.com/pymc-labs/pymc-marketing/assets/59846724/a324b3c7-2bc4-46b6-a7dc-661fcd26900e)

If we switch to:
```python
        gp = pm.gp.HSGP(m=[m], L=[L], cov_func=cov_func)
        phi, sqrt_psd = gp.prior_linearized(Xs=X[:, None] - X_mid)
        hsgp_coefs = pm.Gamma(f"_hsgp_coefs_{name}", size=hsgp_size, alpha=3, beta=7)#, alpha=10, beta=20)
        f = phi @ (hsgp_coefs * sqrt_psd).T
        centered_f = f - f.mean(axis=0) + 1
        return pm.Deterministic(name, centered_f, dims=dims)
```
The prior looks like:
![image](https://github.com/pymc-labs/pymc-marketing/assets/59846724/30adadaa-d9d4-4de0-bfd3-7d20b310edef)

The same happens if we play with a `Beta`
```python
        gp = pm.gp.HSGP(m=[m], L=[L], cov_func=cov_func)
        phi, sqrt_psd = gp.prior_linearized(Xs=X[:, None] - X_mid)
        hsgp_coefs = pm.Beta(f"_hsgp_coefs_{name}", size=hsgp_size, alpha=3, beta=7)#, alpha=10, beta=20)
        f = phi @ (hsgp_coefs * sqrt_psd).T
        centered_f = f - f.mean(axis=0) + 1
        return pm.Deterministic(name, centered_f, dims=dims)
```
![image](https://github.com/pymc-labs/pymc-marketing/assets/59846724/15f6a90a-76dd-41e4-a79a-225dcc0ba1a9)

But everything looks very strange, if we force it to positive with `exp` for example.

![image](https://github.com/pymc-labs/pymc-marketing/assets/59846724/fa651013-424e-4185-abdd-9e7675f302a3)


I am not an expert in HSGPs but from a practical point of view it could be the answer. Maybe @bwengals , @ulfaslak , @juanitorduz who are more involved with HSGPs have the answer?

Does it sound reasonable to be able to allow another distribution that is not Normal? Or it should be normal for some specific property.

Reply:
Indeed just for simplicity, I added `softplus` but not from `pytensor`.

Reply:
@cetagostini I'd be super cautious about changing the distribution of the HSGP coefficients.  The reason their normal is for the "G" in "GP".  I've seen a paper where they use a student-t process and then use a student-t distribution here instead.  The standard way of forcing it positive is to pass it through `pt.exp`, but I suppose softplus would work too.   

If you want to also force it to be zero mean, you can subtract the mean of each basis vector from `phi` directly instead of doing `centered_f = f - f.mean(axis=0)`, which I think would work better.  Either of these will screw up forecasting though, which is the advantage of a GP here over splines or something like that.  I think the best option would be to either:
1. Get rid of the intercept parameter in the model, and just let the HSGP handle it.  If you're transforming it with `pt.exp` it should be able to handle some offsets.  This wouldn't work well if the scales of the intercept and GP were way different.  
2. Keep the intercept but set `drop_first` parameter in HSGP to True.  Depending on your `c` parameter, the first basis vector in `phi` can get very flat and intercept-like.